core/src/main/java/apoc/index/SchemaIndex.java

```diff
@@ -104,35 +124,78 @@ private void scanIndex(BlockingQueue<PropertyValueCount> queue, IndexDefinition
             // we need to using IndexOrder.NONE here to prevent an exception
             // however the index guarantees to be scanned in order unless
             // there are writes done in the same tx beforehand - which we don't do.
-            IndexReadSession indexSession = read.indexReadSession( indexDescriptor );
-            read.nodeIndexScan(indexSession, cursor, IndexQueryConstraints.unorderedValues());
+            
+            final IndexReadSession indexSession;
+            try {
+                indexSession = read.indexReadSession( indexDescriptor );
+            }catch (Exception e) {
+                // we skip indexScan if it's still populating
+                if (e.getMessage().contains("Index is still populating")) {
+                    return;
+                }
+                throw e;
+            }
+            if (isFullText(indexDefinition)) {
+                read.nodeIndexSeek(ktx.queryContext(), indexSession, cursor, IndexQueryConstraints.unconstrained(), PropertyIndexQuery.fulltextSearch("*"));
+            } else {
+                read.nodeIndexScan(indexSession, cursor, IndexQueryConstraints.unorderedValues());
+            }
 
             Value previousValue = null;
             long count = 0;
+            Map<Object, Integer> fullTextMap = new HashMap<>();
             while (cursor.next()) {
-                for (int i = 0; i < cursor.numberOfProperties(); i++) {
-                    Value v = cursor.propertyValue(i);
-                    if (Objects.equals(v, previousValue)) { //  nullsafe equals
-                        count++;
-                    } else {
-                        if (previousValue!=null) {
-                            putIntoQueue(queue, indexDefinition, key, previousValue, count);
+                if (isFullText(indexDefinition)) {
+                    long nodeReference = cursor.nodeReference();
+                    FulltextProcedures.NodeOutput nodeOutput = FulltextProcedures.NodeOutput.forExistingEntityOrNull(tx, nodeReference, cursor.score());
+                    final Node node = nodeOutput.node;
+                    indexDefinition.getPropertyKeys()
+                            .forEach(propKey -> {
+                                final Object property = node.getProperty(propKey, null);
+                                if (property != null) {
+                                    fullTextMap.merge(property, 1, Integer::sum);
+                                }
```

Comment:
I think this is wrong (also see comment in the test). Here you are getting all property values in the index even if you are asking for a specific property in the APOC procedure. 

What makes it even harder is that if you have a node like `(:Label2 {prop1:`BB`})`, that would be in the fulltext index (which have OR semantics) and have the right property but should not be returned by `CALL apoc.schema.properties.distinct(Label1, prop1)` because the node doesn't have Label1.

So we need some sort of filtering on both the label and the property.